### PR TITLE
[vmss] update  and update --protection-policy

### LINKED ISSUE #9164: Support for new parameters for VMSS instances - protectFromScaleIn and protectFromScaleSetActions
**Resource Provider**
CRP

**Description of Feature or Work Requested**
Support for new parameters for VMSS instances - protectFromScaleIn and protectFromScaleSetActions

Under instance properties, the protection parameters are configured as follows:
"protectionPolicy": {
      "protectFromScaleIn": boolean,
      "protectFromScaleSetActions": boolean
    }

This is updateable after create. Not applied at create.

**Minimum API Version Required**
2019-03-01

**Swagger Link**
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json

**Target Date**
05/01/2019
Feature is high-pri ask by external and internal customers

### PR DESCRIPTION
**Pull request by adewaleo**
vmss update now works on vmss instances. Add protection policy arguments for VM instances in a VMSS. Make sure to complete example for vmss update.

closes #9164 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py [hunk 1]
```diff
@@ -1885,11 +1885,18 @@
...
 type: command
-short-summary: Update a VMSS.
+short-summary: Update a VMSS or VM within a VMSS.
 examples:
   - name: Update a VMSS. (autogenerated)
     text: az vmss update --name MyScaleSet --resource-group MyResourceGroup --set virtualMachineProfile.storageProfile.imageReference.version=16.04.201801090
     crafted: true
+  - name: Update a VM instance's protection policies.
+    text: az vmss update --name MyScaleSet --resource-group MyResourceGroup --instance-id 4 \
+            --protection-policy protect_from_scale_in=False protect_from_scale_set_actions=True
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py]: As we discussed, these should be two separate flags, not key-value pairs.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py [hunk 2]
```diff
@@ -1890,6 +1890,15 @@
   - name: Update a VMSS. (autogenerated)
     text: az vmss update --name MyScaleSet --resource-group MyResourceGroup --set virtualMachineProfile.storageProfile.imageReference.version=16.04.201801090
     crafted: true
+  - name: Update a VMSS' license type for Azure Hybrid Benefit.
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py]: `VMSS's`?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 3]
```diff
@@ -343,6 +343,16 @@ def load_arguments(self, _):
         c.argument('accelerated_networking', arg_type=get_three_state_flag(),
                    help="enable accelerated networking. Unless specified, CLI will enable it based on machine image and size")
 
+    with self.argument_context('vmss update') as c:
+        protection_policy_type = CLIArgumentType(overrides=get_three_state_flag(), arg_group="Protection Policy", min_api='2019-03-01')
+
+        c.argument('protect_from_scale_in', arg_type=protection_policy_type, help="Protect the VM instance from scale-in operations.")
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: need `min-api` as well?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py (hunk 2 above)
Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py]: I believe if a word ends with 's', you don't add an extra 's' afterwards.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py (hunk 3 above)
Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: This is specified [here]("/subscriptions/00977cdb-163f-435f-9c32-39ec8ae61f4d/resourceGroups/ova-test/providers/Microsoft.Compute/disks/for-image_OsDisk_1_f06bd58996634ea285226a404eb84aba") in 

```
    with self.argument_context('vmss update') as c:
        protection_policy_type = CLIArgumentType(overrides=get_three_state_flag(), arg_group="Protection Policy", min_api='2019-03-01')

```

**Pull request merged**